quick_start_demo/5-min-Mosaic-AI-Agents-Demo-with-Synthetic-Eval.py

```diff
@@ -0,0 +1,694 @@
+            model_config=agent_config,
+            signature=ModelSignature(
+                inputs=ChatCompletionRequest(),
+                outputs=ChatCompletionResponse(),
+            ),
+            resources=resources,
+            input_example={"messages": [{"role": "user", "content": "What is lakehouse monitoring?"}]},
+            pip_requirements=["databricks-sdk[openai]", "mlflow", "databricks-agents"],
+        )
+
+        # Run evaluation
+        eval_results = mlflow.evaluate(
+            data=evals,  # Your evaluation set
+            model=model_info.model_uri,  # Logged Agent from above
+            model_type="databricks-agent",  # activate Mosaic AI Agent Evaluation
+        )
+
+        return (model_info, eval_results)
+
+
+# Now we call the helper function to run evaluation.
+# The configuration keys must match those defined in `fc_agent.py`
+model_info_gpt4o, eval_results = log_and_evaluate_agent(
+    agent_config={
+        "endpoint_name": "agents-demo-gpt4o",
+        "temperature": 0.01,
+        "max_tokens": 1000,
+        "system_prompt": """You are a helpful assistant that answers questions about Databricks. Questions unrelated to Databricks are irrelevant.
+
+    You answer questions using a set of tools. If needed, you ask the user follow-up questions to clarify their request.
+    """,
+        "max_context_chars": 4096 * 4,
+    },
+    run_name="gpt-4o",
+)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## 3.2/ Compare multiple LLMs on quality/cost/latency
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC **Challenges addressed**
+# MAGIC - How to determine the foundational model that offers the right balance of quality/cost/latency?
+# MAGIC
+# MAGIC **What is happening?**
+# MAGIC
+# MAGIC For the purposes of the demo, let's assume we have fixed any root causes identified above and want to optimize our Agent for quality/cost/latency.  
+# MAGIC
+# MAGIC Here we will run evaluation for several LLMs.  Once evaluation runs, click `View Evaluation Results` to open the MLflow UI for one of the runs.  In the MLFLow Evaluations UI, use the "Compare to Run" dropdown to select another run name.  This comparison view helps you quickly identify where the Agent got better/worse/stayed the same.
+# MAGIC
+# MAGIC Then, go to the MLflow Experiement page and click the chart icon in the upper left corner by `Runs`.  Here, you can compare the models quantiatively across quality/cost/latency metrics.  Cost is proxied through the number of tokens used.
+# MAGIC
+# MAGIC This helps you make informed tradeoffs in partnership with your business stakeholders about cost/latency/quality.  Further, you can use this view to provide quantitative updates to your stakeholders so they can follow your progress improving quality!
+# MAGIC
+# MAGIC ![alt text](https://github.com/databricks/genai-cookbook/blob/demo_gifs/quick_start_demo/demo_gifs/eval_2.gif?raw=true "Title")
+
+# COMMAND ----------
+
+baseline_config = {
```

Comment:
same comments here, need to decide whether or not to remove external models in the notebook